### PR TITLE
DH bindings (#2914)

### PR DESCRIPTION
**Pull request by palaviv**
This PR add binding to the following openssl functions:

- void DH_get0_pqg(const DH *dh, BIGNUM **p, BIGNUM **q, BIGNUM **g)
- int DH_set0_pqg(DH *dh, BIGNUM *p, BIGNUM *q, BIGNUM *g)
- void DH_get0_key(const DH *dh, BIGNUM **pub_key, BIGNUM **priv_key)
- int DH_set0_key(DH *dh, BIGNUM *pub_key, BIGNUM *priv_key)

### CODE REVIEW

**Comment by alex** [comment]
Jenkins, ok to test.

File: src/_cffi_src/openssl/dh.py [hunk 1]
```diff
@@ -26,11 +26,93 @@
 int i2d_DHparams(const DH *, unsigned char **);
 int DHparams_print_fp(FILE *, const DH *);
 int DHparams_print(BIO *, const DH *);
+
+/* added in 1.1.0 when the DH struct was opaqued */
+void DH_get0_pqg(const DH *dh, BIGNUM **p, BIGNUM **q, BIGNUM **g);
+int DH_set0_pqg(DH *dh, BIGNUM *p, BIGNUM *q, BIGNUM *g);
+void DH_get0_key(const DH *dh, BIGNUM **pub_key, BIGNUM **priv_key);
+int DH_set0_key(DH *dh, BIGNUM *pub_key, BIGNUM *priv_key);
```
Comment (alex) [inline src/_cffi_src/openssl/dh.py]: The declerations in this section should not have the variable names here, so `dh`, `pub_key`, `priv_key` etc should be removed here.

**Comment by alex** [comment]
Sorry it took me so long to review this.

File: src/_cffi_src/openssl/dh.py (hunk 1 above)
Comment (palaviv) [inline src/_cffi_src/openssl/dh.py]: done

**Comment by palaviv** [comment]
@alex thats ok we all have our day job :smiley:

**Comment by reaperhulk** [comment]
LGTM. These functions have mutated a bit due to the `const`-ification of getters/setters in recent master but we can wait until that actually appears in a shipped version before making any changes of our own.

**Pull request merged**